mne/viz/_3d.py

```diff
@@ -1036,20 +884,124 @@ def plot_alignment(info=None, trans=None, subject=None, subjects_dir=None,
                               fwd_nn[:, ori, 1],
                               fwd_nn[:, ori, 2],
                               color=color, mode='arrow', scale=1.5e-3)
-    if 'pairs' in fnirs and len(fnirs_picks) > 0:
-        origin = apply_trans(head_trans, np.array(
-            [info['chs'][k]['loc'][3:6] for k in fnirs_picks]))
-        destination = apply_trans(head_trans, np.array(
-            [info['chs'][k]['loc'][6:9] for k in fnirs_picks]))
-        logger.info(f'Plotting {origin.shape[0]} fNIRS pair{_pl(origin)}')
-        renderer.tube(origin=origin, destination=destination)
 
     renderer.set_camera(azimuth=90, elevation=90,
                         distance=0.6, focalpoint=(0., 0., 0.))
     renderer.show()
     return renderer.scene()
 
 
+def _handle_sensor_types(meg, eeg, fnirs):
+    """Handle plotting inputs for sensors types."""
+    if eeg is True:
+        eeg = ['original']
+    elif eeg is False:
+        eeg = list()
+
+    warn_meg = meg is not None  # only warn if the value is explicit
+    if meg is True:
+        meg = ['helmet', 'sensors', 'ref']
+    elif meg is None:
+        meg = ['helmet', 'sensors']
+    elif meg is False:
+        meg = list()
+
+    if fnirs is True:
+        fnirs = ['pairs']
+    elif fnirs is False:
+        fnirs = list()
+
+    if isinstance(meg, str):
+        meg = [meg]
+    if isinstance(eeg, str):
+        eeg = [eeg]
+    if isinstance(fnirs, str):
+        fnirs = [fnirs]
+
+    for kind, var in zip(('eeg', 'meg', 'fnirs'), (eeg, meg, fnirs)):
+        if not isinstance(var, (list, tuple)) or \
+                not all(isinstance(x, str) for x in var):
+            raise TypeError(
+                f'{kind} must be list or tuple of str, got {type(kind)}')
+    for xi, x in enumerate(meg):
+        _check_option(f'meg[{xi}]', x, ('helmet', 'sensors', 'ref'))
+    for xi, x in enumerate(eeg):
+        _check_option(f'eeg[{xi}]', x, ('original', 'projected'))
+    for xi, x in enumerate(fnirs):
+        _check_option(f'fnirs[{xi}]', x, ('channels', 'pairs',
+                                          'sources', 'detectors'))
+    return meg, eeg, fnirs, warn_meg
+
+
+def _plot_sensors(info, to_cf_t, renderer, picks, meg, eeg, fnirs,
+                  warn_meg, head_surf, units, verbose):
+    """Render sensors in a 3D scene."""
+    defaults = DEFAULTS['coreg']
+    ch_pos, sources, detectors = _ch_pos_in_coord_frame(
+        info.pick_channels([info.ch_names[idx] for idx in picks]),
```

Comment:
I feel like I made this comment already... but in any case, `info` needs to be copied here because in `add_sensors` it's not copied and this operates in place. Also `pick_info` is more succinct here (and does a copy=True by default):
```suggestion
        pick_info(info, picks),
```